dvc/dependency/repo.py

```diff
@@ -72,10 +76,42 @@ def fetch(self):
 
         return out
 
+    @staticmethod
+    def _is_git_file(repo, path):
+        if not os.path.isabs(path):
+            try:
+                output = repo.find_out_by_relpath(path)
+                if not output.use_cache:
+                    return True
+            except OutputNotFoundError:
+                return True
+        return False
+
+    def _copy_if_git_file(self, to_path):
+        src_path = self.def_path
+        with self._make_repo(
+            cache_dir=self.repo.cache.local.cache_dir
+        ) as repo:
+            if not self._is_git_file(repo, src_path):
+                return False
+
+            src_full_path = os.path.join(repo.root_dir, src_path)
+            dst_full_path = os.path.abspath(to_path)
+            fs_copy(src_full_path, dst_full_path)
+        return True
+
     def download(self, to):
```

Comment:
The download is not the only place where we need to properly handle git paths. For example see `status` above. Seems like it is going to break on git paths.

Reply:
Should I also add tests for that? I tried to create one based on `test_status.py` (assuming that `dependency.repo.status` is actually called when user runs `dvc status`) but the test file is very minimal and so I didn't find much inspiration there.

Also, since `fetch` is a public method I suppose I will also have to handle git paths in `fetch`. I have the same question re. tests.

Reply:
`.fetch()` is not really public. Generally you only test things that people might access via cli/corresponding `Repo` methods in func tests.

Reply:
I see. Is `.status()` public in this sense?

Reply:
Testing status makes sense, this is not the part of this PR though.

Reply:
`fetch` is fetching cache, which is not really needed for git files, as we clone the git repo anyways. My only concern is the status. To test it, you could run `repo.status("import.dvc")` and check that it is correct.

Reply:
@efiop, could you please provide more context for how to test this?

If I'm to do it programmatically, I don't know how to instantiate and initialize the `repo` object.

I have just tried to test it on the command line (`dvc import` a git-tracked file, and then `dvc status`) but that doesn't produce any issues, as far as I can tell.

Reply:
@Baranowski Good point about it being locked, let's try

```
dvc unlock my.dvc
dvc status my.dvc
```
🙂 

>  I don't know how to instantiate and initialize the repo object.

What do you mean? you are doing it in tests already, no? E.g. `tmp_dir.dvc.imp(str(erepo_dir), src, dst)`, where `tmp_dir.dvc` is a Repo instance.

Reply:
Thanks, @efiop.

now after unlocking and `dvc status`, I get:

```
ERROR: failed to obtain data status - config file error: Section 'remote "auto-generated-upstream"' already exists. Use `-f|--force` to overwrite section with new value.
```
I get the same error when I try to `dvc update` the imported file. Option `-f` or `--force` is not supported for `dvc status` or `dvc update`

Reply:
@Baranowski Indeed, that is a bug introduced by a recently contributed logic. Fixing right now.

Reply:
@Baranowski ah, wait, not able to reproduce it right now. Could you show the test that is failing, please?

Reply:
@efiop, here is a command-line test:
```
  mkdir local_dvc
  cd local_dvc/
  git init
  dvc init
  git commit -av -m "Init DVC"
  cd ../

  mkdir remote_git
  cd remote_git/
  git init
  echo "hello" > file.txt
  git add file.txt 
  git commit -av -m "Init Git"
  dvc init
  git commit -av -m "Init DVC"

  cd ../local_dvc/
  dvc import ../remote_git/ file.txt -o imported_file.txt
  git add imported_file.txt.dvc .gitignore
  git commit -av -m "Imported"
  dvc unlock imported_file.txt.dvc 
  dvc status
```

I also tried to add a unit test (i.e., doing `tmp_dir.dvc.status([...])` after importing a git-tracked file in `tests_import.py`), but it passed without issues.

Reply:
@Baranowski and which `dvc version` are you using in cli?

Reply:
@efiop,

```
$ ../../bin/dvc --version
0.77.3+591560
```

(that's the `dvc` path I was using in the test above, I edited the command lines for clarity)

Reply:
@Baranowski Here is a bit modified script (to not use git files) that works fine for me on 0.77.3:
```
#!/bin/bash

set -e
set -x

rm -rf mytest
mkdir mytest
cd mytest

mkdir local_dvc
cd local_dvc/
git init
dvc init
git commit -av -m "Init DVC"
cd ../

mkdir remote_git
cd remote_git/
git init
echo "hello" > file.txt
#git add file.txt
dvc init
dvc add file.txt
git add file.txt.dvc .gitignore
git commit -av -m "Init Git"
git commit -av -m "Init DVC"

cd ../local_dvc/
dvc import ../remote_git/ file.txt -o imported_file.txt
git add imported_file.txt.dvc .gitignore
git commit -av -m "Imported"
dvc unlock imported_file.txt.dvc
dvc status
```

Feels like there might be an issue with your current implementation, can't put my finger on anything right away.

Reply:
@Baranowski btw, adding `-v` flag to any dvc command will produce a more verbose log (with traceback), which is useful when debugging 🙂

Reply:
@Baranowski See comments below, I think those will fix this issue.

Reply:
I suggest separating this into its own ticket. There are too many questions here:
- we don't really want people make `dvc unlock` to check import status, this makes `DependencyREPO.status()` a half-dead code
- do we want `dvc status` to check import status at all?
- if we do then maybe using `locked` in import implementation is not a good idea
- if it isn't then we need to rework some of our stage code

Reply:
Indeed, just checked our tests and we didn't test status there at all. @Suor Is right, let's create a separate ticket.

Reply:
Opened in https://github.com/iterative/dvc/issues/2959